src/mqt/predictor/rl/Predictor.py

```diff
@@ -32,69 +44,133 @@ def compile_as_predicted(
+        env = rl.PredictorEnv(reward_function)
+        obs = env.reset(benchmark)
+        start_time = time.time()
+        done = False
+        while not done:
+            action_masks = get_action_masks(env)
+            action, _states = model.predict(obs, action_masks=action_masks)
+            action = int(action)
+            obs, reward_val, done, info = env.step(action)
 
-            env = rl.PhaseOrdererEnv(rew)
-            obs = env.reset(file)
-            qc = env.state
-            start_time = time.time()
-            while True:
-                action_masks = get_action_masks(env)
-                action, _states = model.predict(obs, action_masks=action_masks)
-                action = int(action)
-                obs, reward_val, done, info = env.step(action)
-                if done:
-                    duration = time.time() - start_time
-                    if rew == "fidelity":
-                        RL_fid = np.round(
-                            reward.expected_fidelity(env.state, env.device), 2
-                        )
-                        RL_fid_time = np.round(duration, 2)
-                        RL_fid_crit_depth = np.round(reward.crit_depth(env.state), 2)
-                        RL_fid_parallelism = np.round(reward.parallelism(env.state), 2)
-                    elif rew == "parallelism":
-                        RL_parallelism = np.round(reward.parallelism(env.state), 2)
-                        RL_parallelism_time = np.round(duration, 2)
-                        RL_parallelism_fid = np.round(
-                            reward.expected_fidelity(env.state, env.device), 2
-                        )
-                        RL_parallelism_crit_depth = np.round(
-                            reward.crit_depth(env.state), 2
-                        )
-                    elif rew == "critical_depth":
-                        RL_crit_depth = np.round(reward.crit_depth(env.state), 2)
-                        RL_crit_depth_time = np.round(duration, 2)
-                        RL_crit_depth_fid = np.round(
-                            reward.expected_fidelity(env.state, env.device), 2
-                        )
-                        RL_crit_depth_parallelism = np.round(
-                            reward.parallelism(env.state), 2
-                        )
-                    break
+        duration = time.time() - start_time
+        rew_fid = reward.expected_fidelity(env.state, env.device)
+        rew_crit_depth = reward.crit_depth(env.state)
+        rew_gate_ratio = reward.gate_ratio(env.state)
+        rew_mix = reward.mix(env.state, env.device)
+
+        return rl.Result(
+            benchmark=benchmark,
+            used_setup="RL_" + reward_function,
+            time=duration,
+            fidelity=rew_fid,
+            depth=rew_crit_depth,
+            gate_ratio=rew_gate_ratio,
+            mix=rew_mix,
+        )
 
+    elif used_setup == "qiskit_o3":
```

Comment:
I believe this can just be `if` instead of `elif` since you return at the end of the previous statement.

Reply:
done